src/Avalonia.Controls/TransitioningContentControl.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Threading;
+using Avalonia.Animation;
+
+namespace Avalonia.Controls;
+
+/// <summary>
+/// Displays <see cref="Content"/> according to a <see cref="FuncDataTemplate"/>.
+/// Uses <see cref="PageTransition"/> to move between the old and new content values. 
+/// </summary>
+public class TransitioningContentControl : ContentControl
+{
+    private CancellationTokenSource? _lastTransitionCts;
+    private object? _displayedContent;
+    
+    /// <summary>
+    /// Defines the <see cref="PageTransition"/> property.
+    /// </summary>
+    public static readonly StyledProperty<IPageTransition?> PageTransitionProperty =
+        AvaloniaProperty.Register<TransitioningContentControl, IPageTransition?>(nameof(PageTransition),
+            new CrossFade(TimeSpan.FromSeconds(0.5)));
+
+
+    /// <summary>
+    /// Defines the <see cref="DisplayedContent"/> property.
+    /// </summary>
+    public static readonly DirectProperty<TransitioningContentControl, object?> DisplayedContentProperty =
+        AvaloniaProperty.RegisterDirect<TransitioningContentControl, object?>(nameof(DisplayedContent),
+            o => o.DisplayedContent);
+    
+    /// <summary>
+    /// Gets or sets the animation played when content appears and disappears.
+    /// </summary>
+    public IPageTransition? PageTransition
+    {
+        get => GetValue(PageTransitionProperty);
+        set => SetValue(PageTransitionProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets the content displayed whenever there is no page currently routed.
```

Comment:
Readonly property, i think it should be "Gets the content displayed ...".

Reply:
done.